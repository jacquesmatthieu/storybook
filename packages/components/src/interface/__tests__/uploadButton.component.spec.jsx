import React from 'react';
import { shallow } from 'enzyme';

import UploadButton from '../uploadButton.component';

describe('UploadButton', () => {
  let props;

  beforeEach(() => {
    props = {
      text: 'foo',
      icon: 'folder',
      input: {
        name: 'bar',
        onChange: jest.fn(),
      },
    };
  });

  const getWrapper = () => shallow(<UploadButton {...props} />);

  it('should render properly', () => {
    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with value', () => {
    // Given
    props.input.value = [{ name: 'hello' }];

    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });
});
