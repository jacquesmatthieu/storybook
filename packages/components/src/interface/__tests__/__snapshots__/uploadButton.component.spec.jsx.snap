// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadButton should render properly 1`] = `
<InputGroup
  className=""
  fullWidth={false}
  label=""
>
  <FatActionButton
    component={[Function]}
    htmlFor="bar"
    icon="folder"
    text={
      <React.Fragment>
        foo
        <input
          accept="jpg,jpeg,png,pdf"
          hidden={true}
          id="bar"
          name="bar"
          onChange={[Function]}
          type="file"
        />
      </React.Fragment>
    }
  />
</InputGroup>
`;

exports[`UploadButton should render with value 1`] = `
<InputGroup
  className=""
  fullWidth={false}
  label=""
>
  <FatActionButton
    component={[Function]}
    htmlFor="bar"
    icon="folder"
    text={
      <React.Fragment>
        foo
        <input
          accept="jpg,jpeg,png,pdf"
          hidden={true}
          id="bar"
          name="bar"
          onChange={[Function]}
          type="file"
        />
      </React.Fragment>
    }
  />
  <uploadButtoncomponent__FileUploaded
    key="hello"
  >
    <IconButton
      disabled={true}
      hollow={false}
      icon="folder"
      iconProps={Object {}}
      type="button"
    />
    <uploadButtoncomponent__FileUploadedName>
      hello
    </uploadButtoncomponent__FileUploadedName>
    <IconButton
      hollow={true}
      icon="cross"
      iconProps={
        Object {
          "size": 8,
        }
      }
      onClick={[Function]}
      type="button"
    />
  </uploadButtoncomponent__FileUploaded>
</InputGroup>
`;
